### PR TITLE
Fixes #33217 - Add :all_puppetclasses to hostgroup

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
I'm not sure we want to do this. It never was an public API and we are making it an public API by this (it's added to the safemode), I'd rather seen it removed from Host as well, I'm not sure why I've added it (probably as temporary solution).
Could we add it to the Extensions::Hostgroup instead? Ideally for both Host and Hostgroup.

**Comment by xprazak2** [comment]
Updated, methods have been moved to extensions for both Host and Hostgroup.

File: app/models/concerns/foreman_puppet/extensions/host.rb [hunk 1]
```diff
@@ -3,6 +3,8 @@ module Extensions
     module Host
       extend ActiveSupport::Concern
 
+      include ForemanPuppet::Extensions::HostCommon
```
Comment (ezr-ondrej) [inline app/models/concerns/foreman_puppet/extensions/host.rb:6]:
```suggestion
      include ForemanPuppet::Extensions::HostCommon if ForemanPuppet.extracted_from_core?
```

File: app/models/concerns/foreman_puppet/extensions/hostgroup.rb [hunk 2]
```diff
@@ -3,6 +3,8 @@ module Extensions
     module Hostgroup
       extend ActiveSupport::Concern
 
+      include ForemanPuppet::Extensions::HostCommon
```
Comment (ezr-ondrej) [inline app/models/concerns/foreman_puppet/extensions/hostgroup.rb:6]: also only if extracted

**Review by ezr-ondrej — APPROVED** [review]
Seems to work :+1:
Thanks @xprazak2

**Comment by ezr-ondrej** [comment]
CP 1.0: 25a0789

**Pull request merged**